bin/publish.ls

```diff
@@ -0,0 +1,150 @@
+require! {
+  'chalk' : {bold, red, cyan, yellow, green, dim}
+  'inquirer'
+  'jsonfile'
+  'semver'
+  'shelljs/global'
+}
+
+
+SUBPROJECTS_TO_PUBLISH =
+  'exocom-mock-javascript'
+  'exocom-server'
+  'exorelay-javascript'
+  'exoservice-javascript'
+
+# exocom-server is used because there is no root package.json
+# all node exocommunication services are in lock-step versioning
+{name, version} = jsonfile.readFileSync './exocom-server/package.json'
+
+if process.argv.length != 3
+  display-help!
+
+level = process.argv[2]
+display-help! if level in ['-h, --help']
+
+target-version = get-target-version version, level
+
+confirm-target-version ->
+  ensure-no-open-files!
+  ensure-on-master!
+
+  check-npm-dependencies!
+  build-subprojects!
+  run-tests!
+  bump-version-numbers!
+  push-version-numbers!
+  publish-to-npm!
+  push-exocom-docker-image!
+  push-git-tags!
+
+
+function check-npm-dependencies
+  console.log green "Checking npm dependencies...\n"
+  run-command-in-subdirs do
```

Comment:
Proposal for better API of this method:

```ls
run-command-in-subdirs './node_modules/.bin/update-check', message: 'Checking dependencies'
```

As the name of the method implies, the primary and required argument is the command. The second argument are optional options. If no message is given, it prints the command that it runs.
I'd argue that you don't need the message for most runs, since you provide a useful caption in green already.